### PR TITLE
Part of #17670: Replace Typography with Text component in:gas-items-details

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation
This partially fixes [PR](https://github.com/MetaMask/metamask-extension/issues/17670) 
Replaced ```<Typography>``` component with ```<Text>``` for the file gas-details-item-title.js
Also added new stories to Storybook for the files

- ui\components\app\gas-details-item\gas-details-item-title\gas-details-item-title.js
- ui\components\app\gas-details-item\gas-details-item\gas-details-item.js

### New Story for ```gas-details-item.js```
image
image

### New Story for ```gas-details-item-title.js```
image
image

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dhruvv173** [comment]
Hey @georgewrmarshall 
Please review this PR. Replaced Typography with Text component, added new stories and screenshots for the same, if the changes are correct I will go on with making changes to the remaining files.
Thanks!

File: ui/components/app/gas-details-item/gas-details-item-title/gas-details-item-title.js [hunk 1]
```diff
@@ -21,23 +21,23 @@ const GasDetailsItemTitle = () => {
       <InfoTooltip
         contentText={
           <>
-            <Typography variant={TypographyVariant.H7}>
+            <Text variant={TextVariant.bodySm} as="h6">
```
Comment (georgewrmarshall) [inline ui/components/app/gas-details-item/gas-details-item-title/gas-details-item-title.js:24]: Seeing as this text seems to be inside the tooltip could you take a screenshot hovering over the (i) info icon to show the tooltip content

**Review by georgewrmarshall — COMMENTED** [review]
Looking good nice work on the stories! Left one comment

File: ui/components/app/gas-details-item/gas-details-item-title/gas-details-item-title.js (hunk 1 above)
Comment (dhruvv173) [reply ui/components/app/gas-details-item/gas-details-item-title/gas-details-item-title.js:24]: I have added the new screenshots, please review them.
Thanks!

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Thanks for your contribution @dhruvv173

**Review by garrettbear — APPROVED** [review]
Great work and thanks for adding the story!

**Pull request merged**